integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+import com.newrelic.telemetry.spans.SpanBatch;
+import com.newrelic.telemetry.spans.SpanBatchSender;
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import lombok.Value;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.matchers.MatchType;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.MinimumDurationRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+class SpanApiIntegrationTest {
+
+  private static final int SERVICE_PORT = 1080 + new Random().nextInt(900);
+  private static String containerIpAddress;
+  private static MockServerClient mockServerClient;
+  private static final GenericContainer container =
+      new GenericContainer("jamesdbloom/mockserver:mockserver-5.5.1")
+          .withExposedPorts(SERVICE_PORT);
+  private SpanBatchSender spanBatchSender;
+
+  @BeforeAll
+  static void beforeClass() {
+    container.setPortBindings(singletonList(SERVICE_PORT + ":1080"));
+    container.setWaitStrategy(new WaitAllStrategy());
+    container.setStartupCheckStrategy(
+        new MinimumDurationRunningStartupCheckStrategy(Duration.of(10, SECONDS)));
+    container.start();
+    containerIpAddress = container.getContainerIpAddress();
+    mockServerClient = new MockServerClient(containerIpAddress, SERVICE_PORT);
+  }
+
+  @BeforeEach
+  void setUp() throws Exception {
+    mockServerClient.reset();
+    spanBatchSender =
+        SpanBatchSender.builder()
+            .apiKey("fakeKey")
+            .httpPoster(new OkHttpPoster(Duration.ofMillis(1500)))
+            .attributesJson(new AttributesGson(new Gson()))
+            .uriOverride(URI.create("http://" + containerIpAddress + ":" + SERVICE_PORT))
+            .enableAuditLogging()
+            .build();
+  }
+
+  @Test
+  @DisplayName("Low Level SDK can send a span batch and returns a successful response")
+  void testSuccessfulSpanSend() throws Exception {
+    SpanPayload expectedPayload =
+        new SpanPayload(
+            ImmutableMap.of("trace.id", "123456", "attributes", singletonMap("key1", "val1")),
+            singletonList(
+                ImmutableMap.<String, Object>builder()
+                    .put("id", "6f377f46-3f10-4e8e-8390-7717044cdbe9")
+                    .put("trace.id", "123456")
+                    .put("timestamp", 55555)
+                    .put(
+                        "attributes",
+                        ImmutableMap.of(
+                            "duration.ms",
```

Comment:
nitpick: might be nice to see if these pairs will fit on the same line (reads more like mappings that way).

Reply:
fixed